Modify man pages for image-naming

sallyom opened the pull request:
This PR adds recommendations in man pages to use only [a-z0-9-_.] when naming and
tagging images.  The purpose of this is to add consistency and to make
image naming caps rules seem less arbitrary.  This PR addresses confusion with:
1. BaseImage:Tagged (not allowed)
2. baseimage:Tagged (allowed)
3. baseimage/tagged:V1 (allowed)
4. baseimage/Tagged:V1 (not allowed)

Signed-off-by: Sally O'Malley somalley@redhat.com

GordonTheTurtle commented:
Looks like we would not be able to merge this PR because of merge conflicts. Please rebase, fix conflicts, and force push to your branch.

sallyom committed:
Modify man pages for image-naming

This PR adds recommendations in man pages to use only [a-z0-9-_.] when
naming and tagging images.  The purpose of this is to add consistency
and to make image naming caps rules seem less arbitrary.

This PR addresses confusion with:
1. BaseImage:Tagged (not allowed)
2. baseimage:Tagged (allowed)
3. baseimage/tagged:V1 (allowed)
4. baseimage/Tagged:V1 (not allowed)

Signed-off-by: Sally O'Malley <somalley@redhat.com>

moxiegirl commented on man/docker-build.1.md:
```diff
@@ -171,8 +171,8 @@ instruction into the specified target.
-A good practice is to give a name to the image you are building. There are
-no hard rules here but it is best to give the names consideration. 
+A good practice is to give a name to the image you are building. Note that 
+only [a-z0-9-_.] should be used for consistency.  There are no hard rules here but it is best to give the names consideration. 
```
Use only lower `a-z0-9-_.` for consistency.

moxiegirl commented on man/docker-commit.1.md:
```diff
@@ -43,6 +43,9 @@ create a new image run `docker ps` to find the container's ID and then run:
     # docker commit -m="Added Apache to Fedora base image" \
       -a="A D Ministrator" 98bd7fc99854 fedora/fedora_httpd:20
 
+Note that only [a-z0-9-_.] are allowed when naming images from an 
```
ibid. See above

SvenDowideit commented:
LGTM

moxiegirl commented:
LGTM Thanks for the contrib @sallyom

thaJeztah commented:
LGTM, thanks!

sallyom commented:
:)

stevvooe commented on man/docker-build.1.md:
```diff
@@ -171,8 +171,8 @@ instruction into the specified target.
-A good practice is to give a name to the image you are building. There are
-no hard rules here but it is best to give the names consideration. 
+A good practice is to give a name to the image you are building. Note that 
+only a-z0-9-_. should be used for consistency.  There are no hard rules here but it is best to give the names consideration. 
```
> There are no hard rules here but it is best to give the names consideration. 

This is not true, but we've done a lot of work locking this down in the future. A grammar is being added with https://github.com/docker/distribution/issues/690, along with a common package to handle this.
